packages/python/javdb_spider/fetch/fetch_engine.py

```
@@ -376,6 +377,41 @@ def fetch(self, url: str) -> Optional[str]:
+       keeps the spider running when neither name nor URL is configured.
+
+    The ordinal label remains in :attr:`_EngineWorker.proxy_name` for
+    human-readable logs / ban manager / thread name; coordination uses
+    only the value returned here.
+    """
+    name = proxy_config.get('name')
+    if isinstance(name, str) and name.strip():
+        return name.strip()[:256]
+    url = proxy_config.get('https') or proxy_config.get('http')
+    if isinstance(url, str) and url.strip():
+        digest = hashlib.sha1(url.strip().encode('utf-8')).hexdigest()[:16]
```

Ruff (0.15.12)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)